pandas/tseries/offsets.py

```diff
@@ -1039,55 +1039,61 @@ def __init__(self, n=1, normalize=False, weekmask='Mon Tue Wed Thu Fri',
         _CustomMixin.__init__(self, weekmask, holidays, calendar)
 
     @cache_readonly
-    def cbday(self):
-        kwds = self.kwds
-        return CustomBusinessDay(n=self.n, normalize=self.normalize, **kwds)
+    def cbday_roll(self):
+        cbday = CustomBusinessDay(n=self.n, normalize=False, **self.kwds)
+
+        # Define default roll function to be called in apply method
+        if self._prefix.endswith('S'):
+            # MonthBegin
+            roll_func = cbday.rollforward
+        else:
+            # MonthEnd
+            roll_func = cbday.rollback
+        return roll_func
 
     @cache_readonly
     def m_offset(self):
         if self._prefix.endswith('S'):
-            # MonthBegin:
-            return MonthBegin(n=1, normalize=self.normalize)
+            # MonthBegin
+            moff = MonthBegin(n=1, normalize=False)
         else:
             # MonthEnd
-            return MonthEnd(n=1, normalize=self.normalize)
+            moff = MonthEnd(n=1, normalize=False)
+        return moff
 
-
-class CustomBusinessMonthEnd(_CustomBusinessMonth):
-    __doc__ = _CustomBusinessMonth.__doc__.replace('[BEGIN/END]', 'end')
-    _prefix = 'CBM'
+    @cache_readonly
+    def month_roll(self):
+        if self._prefix.endswith('S'):
```

Comment:
can you add a comment to this and cbday_roll. I like the syntax, but a future reader won't get this immediately.